android/library/src/test/java/io/logto/android/auth/browser/BrowserSignInFlowTest.kt

```diff
@@ -13,138 +13,151 @@ import io.logto.client.exception.LogtoException
 import io.logto.client.exception.LogtoException.Companion.EMPTY_REDIRECT_URI
 import io.logto.client.exception.LogtoException.Companion.INVALID_REDIRECT_URI
 import io.logto.client.exception.LogtoException.Companion.MISSING_AUTHORIZATION_CODE
+import io.logto.client.exception.LogtoException.Companion.MISSING_STATE
 import io.logto.client.exception.LogtoException.Companion.SIGN_IN_FAILED
+import io.logto.client.exception.LogtoException.Companion.UNKNOWN_STATE
 import io.logto.client.exception.LogtoException.Companion.USER_CANCELED
 import io.logto.client.model.OidcConfiguration
 import io.logto.client.model.TokenSet
+import io.logto.client.utils.PkceUtils
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.unmockkAll
+import io.mockk.verify
 import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
 import kotlinx.coroutines.ObsoleteCoroutinesApi
+import org.junit.After
 import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.mockito.Mock
-import org.mockito.Mockito.`when`
-import org.mockito.Mockito.anyString
-import org.mockito.Mockito.spy
-import org.mockito.MockitoAnnotations
-import org.mockito.kotlin.anyOrNull
-import org.mockito.kotlin.argumentCaptor
-import org.mockito.kotlin.doAnswer
-import org.mockito.kotlin.eq
-import org.mockito.kotlin.mock
-import org.mockito.kotlin.verify
 import org.robolectric.Robolectric
 import org.robolectric.RobolectricTestRunner
 
 @RunWith(RobolectricTestRunner::class)
 @ObsoleteCoroutinesApi
 @ExperimentalCoroutinesApi
 class BrowserSignInFlowTest {
-    @Mock
+    @RelaxedMockK
     private lateinit var dummyOidcConfiguration: OidcConfiguration
 
-    @Mock
+    @RelaxedMockK
     private lateinit var logtoConfigMock: LogtoConfig
 
-    @Mock
+    @RelaxedMockK
     private lateinit var logtoAndroidClientMock: LogtoAndroidClient
 
-    @Mock
+    @RelaxedMockK
     private lateinit var onCompleteMock: HandleTokenSetCallback
 
     private lateinit var browserSignInFlow: BrowserSignInFlow
 
-    private val logtoExceptionCaptor = argumentCaptor<LogtoException>()
-    private val tokenSetCaptor = argumentCaptor<TokenSet>()
+    private val logtoExceptionSlot = slot<LogtoException>()
 
     @Before
     fun setUp() {
-        MockitoAnnotations.openMocks(this)
+        MockKAnnotations.init(this, relaxUnitFun = true)
 
-        `when`(logtoConfigMock.redirectUri).thenReturn("redirectUri")
+        every { logtoConfigMock.redirectUri } returns "redirectUri"
+        every { logtoAndroidClientMock.logtoConfig } returns logtoConfigMock
 
-        `when`(logtoAndroidClientMock.logtoConfig).thenReturn(logtoConfigMock)
+        mockkObject(PkceUtils)
+        every { PkceUtils.generateState() } returns STATE
```

Comment:
Do not define `STATE` in the companion object of this test class. it's just like a static property which belongs to the class.
We should keep it fresh each time the test class object is instanciated.